### PR TITLE
 Use structured logging in cmd/registry packages

### PR DESCRIPTION
**Pull request by seaneganx**
* Use structured logging
* Replace unchecked errors with explicit discard
* Use `UpdateApiSpec` with `allow_missing` flag in `core/specs.go`
* Formatting adjustments

### CODE REVIEW

File: cmd/registry/controller/local-tasks.go [hunk 1]
```diff
@@ -37,45 +39,46 @@ func (task *ExecCommandTask) String() string {
...
 
 func (task *ExecCommandTask) Run(ctx context.Context) error {
-	//The monitoring metrics/dashboards are built on top of the format of the log messages here.
-	//Check the metric filters before making  any changes to the format.
-	//Location: registry/deployments/controller/dashboard/*
-	taskDetails := fmt.Sprintf("action={%s} taskID={%s}", task.Action.Command, task.TaskID)
+	// The monitoring metrics/dashboards are built on top of the format of the log messages here.
+	// Check the metric filters before making any changes to the format.
+	// Location: registry/deployments/controller/dashboard/*
+	logger := log.WithFields(log.Fields{
```
Comment (shrutiparabgoogle) [inline cmd/registry/controller/local-tasks.go:45]: Just want to call it out here. I am assuming this change works with the dashboard queries that we have.

**Review by shrutiparabgoogle — APPROVED** [review]

File: cmd/registry/controller/local-tasks.go (hunk 1 above)
Comment (seaneganx) [reply cmd/registry/controller/local-tasks.go:45]: These are the dashboard queries I was looking at to maintain backward compatibility.
```
filter: |-
  resource.type="k8s_container"
  resource.labels.project_id="$REGISTRY_PROJECT_IDENTIFIER"
  resource.labels.location="us-central1-a"
  resource.labels.cluster_name="registry-backend"
  resource.labels.namespace_name="default"
  severity>=DEFAULT
  textPayload=~"(Successful|Failed) Execution:"
labelExtractors:
  action: REGEXP_EXTRACT(textPayload, "action={([^})]*)}")
  status: REGEXP_EXTRACT(textPayload, "(Successful|Failed) Execution:")
  task_id: REGEXP_EXTRACT(textPayload, "taskID={([^}]*)}")
```

```
filter: |-
  resource.type="k8s_container"
  resource.labels.project_id="$REGISTRY_PROJECT_IDENTIFIER"
  resource.labels.location="us-central1-a"
  resource.labels.cluster_name="registry-backend"
  resource.labels.namespace_name="default"
  severity>=DEFAULT
  textPayload=~"Generated [0-9.]* actions"
labelExtractors:
  num_tasks: REGEXP_EXTRACT(textPayload, "Generated ([0-9.]*) actions")
```

Here's some example output.
```
 DEBUG[0000] Generating the list of actions... uid=[ 1c0be4fd ]
 DEBUG[0000] Generated 3 actions. Starting Execution... uid=[ 1c0be4fd ]

 DEBUG[0000] Successful Execution:     action={exec echo hello3} taskID={task2} uid=[ 1c0be4fd ]

 DEBUG[0000] Successful Execution:     action={exec echo hello2} taskID={task1} uid=[ 1c0be4fd ]

 DEBUG[0000] Successful Execution:     action={exec echo hello1} taskID={task0} uid=[ 1c0be4fd ]
```

The previous logger had the same format in the parts matched by the dashboard queries.
```
[ e354e13c ] 2021/09/16 16:13:28 Generating the list of actions...
[ e354e13c ] 2021/09/16 16:13:28 Generated 3 actions. Starting Execution...

[ e354e13c ] 2021/09/16 16:13:28 Successful Execution: action={exec echo hello3} taskID={task2}

[ e354e13c ] 2021/09/16 16:13:28 Successful Execution: action={exec echo hello2} taskID={task1}

[ e354e13c ] 2021/09/16 16:13:28 Successful Execution: action={exec echo hello1} taskID={task0}
```

**Comment by seaneganx** [comment]
Force pushed to restart the `cla/google` check that was hanging for several hours.

**Comment by shrutiparabgoogle** [comment]
I realized some of the packages in `cmd/registry ` are still using `fmt.Printf`.
I realized this while running `registry list projects/demo/locations/global/apis/-/versions/-/specs/-`. The output produced by this command doesn't have a `uid` field logged.

 
A quick search gave me the following list where `fmt.Print()` is still being used:
- `registry/cmd/registry/cmd/compute/conformance/api_linter_plugin.go`
- `registry/cmd/registry/cmd/compute/conformance/spectral_linter_plugin.go`
- `registry/cmd/registry/core/print.go`
- `registry/cmd/registry/core/yaml.go`
- `registry/cmd/registry/googleauth/client.go`
- `registry/cmd/registry/main.go`

I am not sure if all of these libraries are used in `registry tool`, might be worth verifying.

**Pull request merged**